emperor/support_files/js/color-view-controller.js

```diff
@@ -174,90 +181,209 @@ define([
   ColorViewController.prototype.constructor = EmperorAttributeABC;
 
 
+  /*
+   * Helper for building the plottables for non-numeric data
+   *
+   * @param {uniqueVals} Array of unique values for the category
+   * @param {data} SlickGrid formatted data from setCategory function
+   *
+   * @return {plottables} Array of plottables for all non-numeric values
+   *
+   */
+   ColorViewController._nonNumericPlottables = function(uniqueVals, data) {
+     // Filter down to only non-numeric data
+     var nonNumeric = uniqueVals.filter(isNaN);
+     var plotList = data.filter(function(x) {
+       return $.inArray(x.category, nonNumeric) !== -1;
+     });
+     // Build list of plottables and return
+     var plottables = [];
+     for (var i = 0; i < plotList.length; i++) {
+       plottables = plottables.concat(plotList[i].plottables);
+     }
+     return plottables;
+   }
+
+
   /**
    *
-   * Generate a list of colors that corresponds to all the samples in the plot
+   * Wrapper for generating a list of colors that corresponds to all samples
+   * in the plot by coloring type requested
    *
-   * @param {values} list of objects to generate a color for, usually a
+   * @param {Array} [values] list of objects to generate a color for, usually a
    * category in a given metadata column.
-   * @param {map} name of the color map to use, see
+   * @param {String} [map] name of the color map to use, see
    * ColorViewController.Colormaps
-   * @param {discrete} Whether to treat colormap requested as a discrete set
-   * of colors or use interpolation to create gradient of colors
-   * @param {scaled} Whether to use a scaled colormap or equidistant colors for
-   * each value. Default is false.
+   * @param {Boolean} [discrete] Whether to treat colormap requested as a
+   * discrete set of colors or use interpolation to create gradient of colors
+   * @param {Boolean} [scaled] Whether to use a scaled colormap or equidistant
+   * colors for each value. Default is false.
    *
+   * @return {Object} [colors] The object containing the hex colors keyed to
+   * each sample
+   * @return {String} [gradientSVG] The SVG string for the scaled data or null
    *
-   * This function will generate a list of coloring values depending on the
-   * coloring scheme that the system is currently using (discrete or
-   * continuous).
    */
   ColorViewController.getColorList = function(values, map, discrete, scaled) {
-    var colors = {}, numColors = values.length - 1, counter = 0;
-    var interpolator, min, max;
-    var scaled = scaled || false;
+    var colors = {}, gradientSVG;
+    scaled = scaled || false;
```

Comment:
yes, it makes the coloring default to not showing scaled bar if called, which should happen.